### ISSUE TITLE
CI: build fails on nixos-unstable (Rust 1.80 broke the `time` crate)

### ISSUE THREAD

**Issue by 0xB10C** [0xB10C/nix#40]
https://github.com/0xB10C/nix/actions/runs/10416005091/job/28861536640#step:8:1898 errors:

error (ignored): error: cannot unlink '/tmp/nix-build-bitcoind-v27.1.drv-0': Directory not empty

**Comment by 0xB10C** [comment]
The actual error seems to be: (found with #42)

```
checking for stdint.h... yes
checking for strings.h... yes
   Compiling data-encoding v2.5.0
checking for sys/stat.h... yes
checking for sys/types.h... yes
error: builder for '/nix/store/1xyz1rnq9cr13kcrz5kyxhcrhbw3n49w-github-metadata-backup.drv' failed with exit code 101;
       last 25 log lines:
       >    Compiling percent-encoding v2.3.0
       >    Compiling serde_json v1.0.104
       >    Compiling form_urlencoded v1.2.0
       >    Compiling addr2line v0.20.0
       >    Compiling time v0.3.25
       >    Compiling anstyle-parse v0.2.1
       >    Compiling miniz_oxide v0.7.1
       > error[E0282]: type annotations needed for `Box<_>`
       >   --> /build/github-metadata-backup-vendor.tar.gz/time/src/format_description/parse/mod.rs:83:9
       >    |
       > 83 |     let items = format_items
       >    |         ^^^^^
       > ...
       > 86 |     Ok(items.into())
       >    |              ---- type must be known at this point
       >    |
       > help: consider giving `items` an explicit type, where the placeholders `_` are specified
       >    |
       > 83 |     let items: Box<_> = format_items
       >    |              ++++++++
       >
       >    Compiling hyper v0.14.27
       > For more information about this error, try `rustc --explain E0282`.
       > error: could not compile `time` (lib) due to 1 previous error
       > warning: build failed, waiting for other jobs to finish...
       For full logs, run 'nix log /nix/store/1xyz1rnq9cr13kcrz5kyxhcrhbw3n49w-github-metadata-backup.drv'.
error (ignored): error: cannot unlink '/tmp/nix-build-bitcoind-v27.1.drv-0/source/src/secp256k1': Directory not empty
error: build of '/nix/store/1xyz1rnq9cr13kcrz5kyxhcrhbw3n49w-github-metadata-backup.drv', '/nix/store/8lr1ayxdyrhycfvfns9h7qxsic4l60dr-stratum-observer.drv', '/nix/store/clh31dd4kybjw5kbcdv76xlx6j074irg-addrman-observer-proxy.drv', '/nix/store/kifivbhhyfsfadjyvy11bbjpdxbd0wss-bitcoind-v27.1.drv', '/nix/store/p6l5nl4i594r1z0kac0jkbw2q9vrfpd7-miningpool-observer.drv', '/nix/store/p9c1h1597mg4v0cnzr1mjqmcdnn03abx-peer-observer.drv', '/nix/store/pv9czyj9lwh4drfks26xff6rn2g9186p-fork-observer.drv', '/nix/store/r2s1fw3rniacj2mh9vh9m22jhhs8p483-transactionfee-info.drv' failed
Error: Process completed with exit code 1.
```

Rust 1.80.0 breaks old versions of the time package.

https://discourse.nixos.org/t/rustc-explain-e0282-error-could-not-compile-time-lib/50765/2

**Comment by 0xB10C** [comment]
https://github.com/0xB10C/github-metadata-backup/pull/4

**Comment by 0xB10C** [comment]
also, https://github.com/0xB10C/miningpool-observer/pull/82

**Comment by 0xB10C** [comment]
> also, [0xB10C/miningpool-observer#82](https://github.com/0xB10C/miningpool-observer/pull/82)

The miningpool-observer CI needs to wait for Rust 1.78 which will be available with NixOS 24.11.

**Comment by 0xB10C** [comment]
reopend a PR that actually builds, but this is still blocked on NixOS 24.11

https://github.com/0xB10C/miningpool-observer/pull/83#issuecomment-2451703231
